releasenotes/notes/fix-filetype-router-inconsistencies-b22a3af00059f953.yaml

```
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    `FileTypeRouter` has a new parameter `raise_on_failure` with default value to `False`. When set to `True`, `FileNotFoundError` is always raised for non-existent files.
+    Previously, this exception was raised only when processing a non-existent file and the `meta` parameter was provided to `run()`.
+enhancements:
+  - |
+    `MultiFileConverter` outputs a new key `failed` in the result dictionary, which contains a list of files that failed to convert.
```

```suggestion
upgrade:
  - |
    `MultiFileConverter` outputs a new key `failed` in the result dictionary, which contains a list of files that failed to convert.
    The `documents` output is only included if at least one file is successfully converted. Previously, `documents` was always present, even when empty.
```

I would expand a bit the part where we explain about `documents` output. I would also put this in `upgrade` to make it more prominent.